File: ray-tracing-web/src/source/surfaces/Shiny.ts
import { Color } from '../Color';
import { Vector } from '../Vector';
import { Surface } from '../Surface';

export class Shiny implements Surface {
    public readonly roughness = 250

    public diffuse(pos: Vector): Color {
        return Color.white;
    }

    public specular(pos: Vector): Color {
        return Color.grey;
    }

    public reflect(pos: Vector): number {
        return 0.7;
    }
}

File: ray-tracer/elements/surfaces/Shiny.cs
using System;
using ray_tracer.common;
using ray_tracer.scene;


namespace ray_tracer.elements.surfaces
{
    public class Shiny : Surface
    {
        public double roughness => 250.0;

        public Color diffuse(Vector pos)
        {
            return Color.white;
        }

        public double reflect(Vector pos)
        {
            return 0.7;
        }

        public Color specular(Vector pos)
        {
            return Color.grey;
        }
    }
}
